### ISSUE TITLE
Search engine icon in search bar' script no longer works in Firefox 77

### ISSUE THREAD

**Issue by anomiex** [Aris-t2/CustomJSforFx#33]
Looks like they removed the `setIcon()` call this script was hooking into in https://hg.mozilla.org/releases/mozilla-release/rev/68d0f6719da53dd6b0bbed26073560bf2627e53d

We could easily enough hook the `updateDisplay()` call that had been calling `setIcon()` instead.

**Comment by Speravir** [comment]
@anomiex @Aris-t2 The script with this new code still does not work for me, but the one declared _old_ (`alternative_searchbar_fx75-77_old.uc.js`) with the fix by Aris works fine here, both with the same delay value. Am I still supposed to adjust the initialisation delay despite the new addings, i.e. I have to test by increasing it further?

**Comment by Aris-t2** [comment]
@Speravir 

Could you setup "Portable Firefox" in a new folder, use "method 2" for the JS part and only test [search_engine_icon_in_searchbar.uc.js](https://github.com/Aris-t2/CustomJSforFx/blob/master/scripts/search_engine_icon_in_searchbar.uc.js) file. Despite the result repeat the same with [alternative_searchbar.uc.js](https://github.com/Aris-t2/CustomJSforFx/blob/master/scripts/alternative_searchbar.uc.js).

There must be a reason why the scripts currently do not work for you. I suspect a conflict with another script.

**Comment by Speravir** [comment]
> Could you setup "Portable Firefox" in a new folder, use "method 2" for the JS part

I did it slightly different with the userChrome,js setting from Camp Firefox. What I found out: In a new profile _all_ of my user scripts work together with alternative searchbar script _without_ this one found in Ardiman’s resource: [restartpanelbutton](https://github.com/ardiman/userChrome.js/tree/master/Restart_Panelbutton): enabled – searchbar script will not work, disabled – it will work. I could live without it, because I have another script, [restartfirefox](https://github.com/ardiman/userChrome.js/tree/master/RestartFirefox), in use which in fact I use much more often.

But: In my real profile it did not help to remove the panelbutton script, so I will have to investigate further, and this will take some time.

**Comment by Aris-t2** [comment]
Note:
[restart_item_in_menu.uc.js](https://github.com/Aris-t2/CustomJSforFx/blob/master/scripts/restart_item_in_menu.uc.js) offers a restart item inside panel/menu button and inside menubars "File" menu.

[restart_button.uc.js](https://github.com/Aris-t2/CustomJSforFx/blob/master/scripts/restart_button.uc.js) adds a toolbar button.

Both showed no issues when used together with other scripts. Maybe that is something for a future setup or yours.

**Comment by Speravir** [comment]
First side remark:
> Both showed no issues when used together with other scripts. Maybe that is something for a future setup or yours.

Yes, I noticed them myself and will change to them.

But now for the embarrassing part (blushing) – I wrote:
> In my real profile it did not help to remove the panelbutton script,

In my profile I have adjusted the delay time in *alternative_searchbar.uc.js*, but I have _also_ added a note for myself that the variable `searchsettingslabel` is not used anymore for FF versions above v.70. This note is in German containing an umlaut. And I unintentionally saved in Windows encoding, **not in UTF-8**! Therefore the file wasn’t read at all. After fixing this all works fine (with my individual delay)! (And this issue ticket could be closed.)

Given that this script is thought to be the one for recent Firefox versions perhaps this variable and related code could be removed? In active script versions this would be line 49 and adjustments in lines 156–159.
